pkg/common/commands/resetoffset/controller/offsets.go

```diff
@@ -192,42 +208,28 @@ func safeCloseSaramaClient(logger *zap.Logger, client sarama.Client) {
 	}
 }
 
-// safeCloseOffsetManager wraps the Sarama OffsetManager.Close() call for safe usage as a defer statement
-func safeCloseOffsetManager(logger *zap.Logger, offsetManager sarama.OffsetManager) {
-	if offsetManager != nil {
-		err := offsetManager.Close() // This is fast ; )
-		if err != nil {
-			logger.Error("Failed to close Sarama OffsetManager", zap.Error(err))
-		}
-	}
-}
-
-// closePartitionOffsetManager wraps the Sarama OffsetManager.Close() call with nil check and error logging
-func closePartitionOffsetManager(logger *zap.Logger, partitionOffsetManager sarama.PartitionOffsetManager) {
-	if partitionOffsetManager != nil {
-		err := partitionOffsetManager.Close() // Takes several seconds in success case & can hang (on AsyncClose lock) if ConsumerGroups not stopped!
-		if err != nil {
-			logger.Error("Failed to close Sarama PartitionOffsetManager", zap.Error(err))
-		}
-	}
-}
-
-// drainPartitionOffsetManagerErrors drains the PartitionOffsetManager's Error channel and returns any ConsumerErrors
-func drainPartitionOffsetManagerErrors(logger *zap.Logger, partitionOffsetManager sarama.PartitionOffsetManager) sarama.ConsumerErrors {
-	consumerErrs := make(sarama.ConsumerErrors, 0)
-	if partitionOffsetManager != nil {
-		select {
-		case consumerErr, ok := <-partitionOffsetManager.Errors():
-			if consumerErr != nil {
-				logger.Error("Failed to mark/reset Offset", zap.Error(consumerErr.Unwrap()))
-				consumerErrs = append(consumerErrs, consumerErr)
-			}
-			if !ok {
-				break // Error Channel Closed - Stop Draining
+// drainPartitionOffsetManagerErrors drains the PartitionOffsetManager's Error channels and
+// returns any ConsumerErrors as a Zap multierr, and must be called after Commit() / Close().
+func drainPartitionOffsetManagerErrors(partitionOffsetManagers ...sarama.PartitionOffsetManager) error {
+	var multiErr error
+	for _, partitionOffsetManager := range partitionOffsetManagers {
+		if partitionOffsetManager != nil {
+			select {
+			case consumerErr, ok := <-partitionOffsetManager.Errors():
+				if consumerErr != nil {
+					if multiErr == nil {
+						multiErr = consumerErr.Unwrap()
+					} else {
+						multierr.AppendInto(&multiErr, consumerErr.Unwrap())
+					}
+				}
+				if !ok {
+					break // Error Channel Closed - Stop Draining
+				}
+			case <-time.After(5 * time.Second):
```

Comment:
This is giving each POM five full seconds before moving to the next one, right ?  I don't know how many partitions are likely to be in play here but that could add up to a fair bit of sleep-time if they're all timing out.  That may not actually matter but I thought I'd mention it.

Reply:
Yeah, that is the error/edge case and normal execution should hit the other case: statement.  I could reduce it slightly but seems safer to err on the side of caution here. going to leave as is - but good question.